1_fetch/src/fetch_wqp_data.R

```diff
@@ -184,10 +189,25 @@ fetch_wqp_data <- function(site_counts_grouped, characteristics, wqp_args = NULL
                            characteristicName = c(characteristics)))
   }
   
-  # Pull data, retrying up to the number of times indicated by `max_tries`
-  wqp_data <- retry::retry(dataRetrieval::readWQPdata(wqp_args_all),
-                           when = "Error:", 
-                           max_tries = max_tries)
+  # Define function to pull data, retrying up to the number of times
+  # indicated by `max_tries`
+  pull_data <- function(x){
+    retry::retry(dataRetrieval::readWQPdata(x),
+                 when = "Error:", 
+                 max_tries = max_tries)
+  }
+  
+  # Now pull the data. Capture messages returned from the call to readWQPdata. 
+  # If verbose == TRUE, print all captured messages.
+  messages <- list()
+  wqp_data <- withCallingHandlers({
+    pull_data(wqp_args_all)
+    },
+    message = function(msg) {
+      messages <<- c(messages, msg)
+      tryInvokeRestart('muffleMessage')
+    })
+  if(verbose) message(messages)
```

Comment:
I don't really have a good reason here. I was hesitant to use a blanket `suppressMessages()` wrapper around `pull_data()` in case there were other messages that would be unintentionally suppressed. However, my implementation using `withCallingHandlers()` would result in the same outcome, it just stashes those messages instead of suppressing them. Your implementation is simpler, so I say we go with that!

Reply:
Ok, I see. So you _could_ setup fancy handling with your approach because we are keeping the messages in `messages`. We could use that to only exclude the messages related to the empty data, but you are not currently doing that so I think this basic handling would be sufficient!

Reply:
Yes, exactly! I was inspired to play with this for 10 minutes and it looks like I _could_ set up the code so that we suppress or "muffle" messages only if they are associated with those 'no data' messages (commit https://github.com/USGS-R/ds-pipelines-targets-example-wqp/pull/66/commits/1f211f9f4c61900cafda82c9795b9e2112e3d6d3):

```r
  messages <- list()
  wqp_data <- withCallingHandlers({
    pull_data(wqp_args_all)
    },
    message = function(msg) {
      messages <<- c(messages, msg$message)
      if(startsWith(conditionMessage(msg), "The following url returned no data")|
         startsWith(conditionMessage(msg), "https://")){
        invokeRestart('muffleMessage')}
    })
  if(verbose) message(messages)
```

It's nice to know we can do this! However, there's still good reason to use basic handling here IMO: 

- I don't have a specific use case in mind for other dataRetrieval messages that we would want to get passed up even if `verbose = FALSE`. 
- I also hope that the code in this pipeline is readily interpretable for most users. I expect more users will be familiar with the basic handling approach in  https://github.com/USGS-R/ds-pipelines-targets-example-wqp/pull/66/commits/dff5ce9eb5be2bccf5d87666e54c92ff76929a96 compared to condition handling using `withCallingHandlers()`.